common/conf/conf.cpp

```diff
@@ -124,6 +127,69 @@ ExpectedOptionValue CmdlineOptionsIterator::Next() {
 	return ExpectedOptionValue({std::move(option), std::move(value)});
 }
 
+error::Error MenderConfig::ProcessCmdlineArgs(const vector<string> &args) {
+	string config_path = paths::DefaultConfFile;
+	string fallback_config_path = paths::DefaultFallbackConfFile;
+	string log_file = "";
+	string log_level = log::ToStringLogLevel(log::kDefaultLogLevel);
+
+	CmdlineOptionsIterator opts_iter(
+		args,
+		{"--config",
+		 "-c",
+		 "--fallback-config",
+		 "-b",
+		 "--data",
+		 "-d",
+		 "--log-file",
+		 "-L",
+		 "--log-level",
+		 "-l"},
+		{});
+	auto ex_opt_val = opts_iter.Next();
+	while (ex_opt_val && ((ex_opt_val.value().option != "") || (ex_opt_val.value().value != ""))) {
+		auto opt_val = ex_opt_val.value();
+		if ((opt_val.option == "--config") || (opt_val.option == "-c")) {
+			config_path = opt_val.value;
+		} else if ((opt_val.option == "--fallback-config") || (opt_val.option == "-b")) {
+			fallback_config_path = opt_val.value;
+		} else if ((opt_val.option == "--data") || (opt_val.option == "-d")) {
+			data_store_dir = opt_val.value;
+		} else if ((opt_val.option == "--log-file") || (opt_val.option == "-L")) {
+			log_file = opt_val.value;
+		} else if ((opt_val.option == "--log-level") || (opt_val.option == "-l")) {
+			log_level = opt_val.value;
+		}
+	}
+	if (!ex_opt_val) {
+		return ex_opt_val.error();
+	}
+
+	if (log_file != "") {
+		auto err = log::SetupFileLogging(log_file, true);
+		if (err) {
+			return err;
+		}
+	}
+
+	auto ex_log_level = log::StringToLogLevel(log_level);
+	if (!ex_log_level) {
+		return ex_log_level.error();
+	}
+	SetLevel(ex_log_level.value());
+
+	auto ret = this->LoadFile(fallback_config_path);
+	if (!ret || !this->ValidateConfig()) {
+		this->Reset();
+	}
+	ret = this->LoadFile(config_path);
+	if (!ret || !this->ValidateConfig()) {
+		this->Reset();
+	}
```

Comment:
Shouldn't we return error here?

Reply:
I don't know which cases are errors, TBH. There are the cases in which I would return an error:
- failing to load the config given as a cmdline option, missing or invalid (because it was explicitly specified)
- failing to load the fallback config given as a cmdline option, missing or invalid (because it was explicitly specified)
- present but invalid config at the default path
- present but invalid fallback config at the default path

Do we have a spec of what the current Go-based implementation does?

Reply:
The Golang client returns errors for all of those cases, apparently.